src/desdemona/query.clj

```diff
@@ -53,3 +61,24 @@
    (run-logic-query (dsl->logic dsl-query) events))
   ([n-answers dsl-query events]
    (run-logic-query n-answers (dsl->logic dsl-query) events)))
+
+(def ^:private infix-parser
+  (insta/parser (resource "infix-query-grammar.ebnf")))
+
+(defn ^:private parsed-infix->dsl
+  [parsed]
+  (m/match parsed
+    [:expr terms]
+    (parsed-infix->dsl terms)
+
+    [:eq
+     [:fn-call
+      [:identifier "ip"]
+      [:identifier arg]]
+     [:ipv4-address & addr-parts]]
+    (let [arg (symbol arg)
+          addr (s/join "." addr-parts)]
+      `(~'= (:ip ~arg) ~addr))))
```

Comment:
I think that makes sense. Mostly just commenting on "it seems like we're explicitly only supporting IP addresses here, for now". I don't think we need to change that.